solvers/DeepCORAL.py

```diff
@@ -0,0 +1,124 @@
+from benchopt import safe_import_context
+
+# Protect the import with `safe_import_context()`. This allows:
+# - skipping import to speed up autocompletion in CLI.
+# - getting requirements info when all dependencies are not installed.
+with safe_import_context() as import_ctx:
+    from benchmark_utils.base_solver import DASolver
+    from skada.deep import DeepCoral
+    import torch
+    from torch import nn
+    from torch.optim import AdamW
+    from skada.metrics import SupervisedScorer, DeepEmbeddedValidation
+
+
+class ShallowConvNet(nn.Module):
```

Comment:
These networks are not specific to DeepCoral. Maybe we can create a utils/architectures.py